l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -278,51 +359,89 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         returns (L2TransactionRequestTwoBridgesInner memory request)
     {
         require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
+        bytes32 _assetId;
+        bytes memory _transferData;
+        bool legacyDeposit = false;
+        try this.handleLegacyData(_data, _prevMsgSender) returns (
```

Comment:
Why `_data` decoding for `address, uint256, address` can't have a collision with `bytes32, bytes`?

Reply:
`try`/`catch` is an absolute evil, it fail for unforseen reason (gas call in erc20 ?) and you will never know what exactly happen.

Then encoding can have collision, not good too. 

My proposal would be create unique encoding and replace try/catch with if/else!

The encoding we can use:
legacy: `abi.encode(address, uint256, address)`
new: `0x01 || abi.encode(bytes32, bytes)`

Since `0x01` can never be the first byte in legacy encoding (address consume the last 20 bytes). it is unique encoding.